RecoMuon/MuonIdentification/src/MuonShowerInformationFiller.cc

```diff
@@ -276,36 +276,7 @@ MuonShowerInformationFiller::hitsFromSegments(const GeomDet* geomDet,
 //
 // Find cluster
 //
-MuonTransientTrackingRecHit::MuonRecHitContainer 
-MuonShowerInformationFiller::findPhiCluster(MuonTransientTrackingRecHit::MuonRecHitContainer& muonRecHits, 
-                              const GlobalPoint& refpoint) const {
-
-  if ( muonRecHits.empty() ) return muonRecHits;
-
-  //clustering step by phi
-  float step = 0.05;
-  MuonTransientTrackingRecHit::MuonRecHitContainer result;
-
-  stable_sort(muonRecHits.begin(), muonRecHits.end(), AbsLessDPhi(refpoint));
 
-  for (MuonTransientTrackingRecHit::MuonRecHitContainer::const_iterator ihit = muonRecHits.begin(); ihit != muonRecHits.end() - 1; ++ihit) {
-      if (fabs(deltaPhi((*(ihit+1))->globalPosition().phi(), (*ihit)->globalPosition().phi() )) < step) {

```

Comment:
For the changes to use the simplified vector instead of the plain rechits, I meant all of the code (not just the for loop).
The sorting issue will not go away; the code will be much faster though.